common/src/com/thoughtworks/go/config/materials/perforce/P4Material.java

```diff
@@ -109,6 +109,11 @@ private P4Material(String serverAndPort, String view, String userName, String fo
     }
 
     @Override
+    public void checkout(File baseDir, Revision revision, SubprocessExecutionContext execCtx) {
+        throw new RuntimeException("Checkout not implemented");
```

Comment:
Any specific reason why this throws not implemented exception? Wouldn't delegating it to `updateTo` like the other scms be enough?